File: MoveZeroes.c
void moveZeroes(int* nums, int numsSize) {
    int i = 0;
    int j;
    for (j = 0; j < numsSize; ++j) {
        if (*(nums + j)) {
            *(nums + i++) = *(nums + j);
        }
    }
    while (i < numsSize) {
        *(nums + i++) = 0;
    }
}

File: MoveZeroes.py
class Solution(object):
    def moveZeroes(self, nums):
        """
        :type nums: List[int]
        :rtype: void Do not return anything, modify nums in-place instead.
        """
        i = 0
        for elt in nums:
            if elt != 0:
                nums[i] = elt
                i += 1
        while i < len(nums):
            nums[i] = 0
            i += 1